src/sdam/test.rs

```diff
@@ -451,3 +451,35 @@ async fn hello_ok_true() {
             .expect("subsequent heartbeats should use hello");
     }
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn repl_set_name_mismatch() -> crate::error::Result<()> {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+    if !client.is_replica_set() {
+        log_uncaptured("skipping repl_set_name_mismatch due to non-replica set topology");
+        return Ok(());
+    }
+
+    let mut options = CLIENT_OPTIONS.clone();
+    options.hosts.drain(1..);
+    options.direct_connection = Some(true);
+    options.repl_set_name = Some("invalid".to_string());
+    let client = Client::with_options(options)?;
```

Comment:
This needs to be a `Client` rather than a `TestClient` because the latter executes a few operations with `unwrap()`s, which fail and crash before the test can check anything.